src/libraries/System.Net.Mail/src/System/Net/Mail/Attachment.cs

```diff
@@ -204,7 +204,7 @@ public string ContentId
                     ContentId = cid;
                     return cid;
                 }
-                if (cid.Length >= 2 && cid[0] == '<' && cid[cid.Length - 1] == '>')
+                if (cid.StartsWith('<') && cid.EndsWith('>'))
```

Comment:
In MailAddress (the next file) you left the Length check, but you removed it here.

Feels like same shape before should have same shape after (either all keep or all cut the length check).